gapis/api/vulkan/replay.go

```diff
@@ -49,59 +49,77 @@ var (
 	_ = replay.Profiler(API{})
 )
 
-// GetReplayPriority returns a uint32 representing the preference for
-// replaying this trace on the given device.
-// A lower number represents a higher priority, and Zero represents
-// an inability for the trace to be replayed on the given device.
-func (a API) GetReplayPriority(ctx context.Context, i *device.Instance, h *capture.Header) uint32 {
+// GetReplayPriority implements the replay.Support interface
+func (a API) GetReplayPriority(ctx context.Context, i *device.Instance, h *capture.Header) (uint32, device.Incompatibility) {
 	devConf := i.GetConfiguration()
 	devAbis := devConf.GetABIs()
 	devVkDriver := devConf.GetDrivers().GetVulkan()
 	traceVkDriver := h.GetDevice().GetConfiguration().GetDrivers().GetVulkan()
 
+	// Trace has no Vulkan information
 	if traceVkDriver == nil {
-		log.E(ctx, "Vulkan trace does not contain VulkanDriver info.")
-		return 0
+		log.E(ctx, "Vulkan trace does not contain VulkanDriver info")
+		return 0, device.Incompatibility_NoTraceVkDriver
 	}
 
 	// The device does not support Vulkan
 	if devVkDriver == nil {
-		return 0
+		return 0, device.Incompatibility_NoVkSupport
 	}
 
+	var incompatibility device.Incompatibility
 	for _, abi := range devAbis {
```

Comment:
Since here we iterate through all the ABIs, the returned incompatibility reason might only record the reason found in the last ABI. it might be a nit improvement, but maybe we could rank the incompatibility reason and return the most detailed one throughout the iteration?

Reply:
Good point! I added a "reasonDepth" check such that we retain the incompatibility reason with the deepest depth, i.e. the most detailed one. I hope this covers your remark.